test/run-end-to-end.py

```diff
@@ -104,10 +104,18 @@ def do_test():
     #
     # Check that slot reconciliation is working on agent
     # 
+    temporary_agent_log_path = "/tmp/agent_log.txt"
     print("Check logs for Agent slot-reconciliation for pod {}".format(shared_test_code.agent_pod_name))
-    result = os.system('sudo {} logs {} | grep "get_node_slots - crictl called successfully" | wc -l | grep -v 0'.format(kubectl_cmd, shared_test_code.agent_pod_name))
-    if result != 0:
-        print("Akri failed to successfully connect to crictl via the CRI socket")
+    for x in range(3):
+        log_result = os.system('sudo {} logs {} >> {}'.format(kubectl_cmd, shared_test_code.agent_pod_name, temporary_agent_log_path))
+        if log_result == 0:
+            break
+        print("Failed to get logs from {} pod with result {} on attempt {} of 3".format(shared_test_code.agent_pod_name, log_result, x))
+        if x == 3:
+            return False
+    grep_result = os.system('grep "get_node_slots - crictl called successfully" {} | wc -l | grep -v 0'.format(temporary_agent_log_path))
+    if grep_result != 0:
+        print("Akri failed to successfully connect to crictl via the CRI socket with stdout result of {}", grep_result)
```

Comment:
maybe include a `os.system('sudo {} get pods,services,akric,akrii --show-labels'.format(kubectl_cmd))` here?

Reply:
(so that if there is an error, we can see if the desired pod is running)

Reply:
maybe we should call `os.system('grep "get_node_slots" {}'.format(temporary_agent_log_path))` in the error condition too, just to try to see what is going on when this fails.

Reply:
From what i understand, `os.system` does not output anything. It just returns the return code. But we can use the [`subprocess` lib](https://stackoverflow.com/questions/4760215/running-shell-command-and-capturing-the-output) to output the results.

Reply:
i think we see the output of the helm install command and it seems to use os.system

```
    helm_install_command = "helm install akri {} --set debugEcho.enabled=true --set debugEcho.name={} --set debugEcho.shared=false --set agent.allowDebugEcho=true {} {} --debug ".format(helm_chart_location, shared_test_code.DEBUG_ECHO_NAME, cri_args, extra_helm_args)
    print("Helm command: {}".format(helm_install_command))
    os.system(helm_install_command)
```

Reply:
Was overthinking it. You're right. Thanks!